templates/prompt-for-mfa-u2f.php

```diff
@@ -4,20 +4,66 @@
 $this->includeAtTemplateBase('includes/header.php');
 
 ?>
-<form action="<?= htmlentities($this->data['formTarget']); ?>">
+<script src="<?=SimpleSAML_Module::getModuleURL('mfa/u2f-api.js');?>"></script>
+<script type="application/javascript">
+    window.onload = function() {
+
+      console.log('data:');
+      console.log(<?=json_encode($this->data)?>);
+
+      var mfa = <?php echo json_encode($this->data['mfaOption']['data'], JSON_PRETTY_PRINT);?>;
+      console.log(mfa);
+      u2f.sign(mfa.appId, mfa.challenge, [mfa], function(response) {
+        console.log(response);
+        if (response.errorCode && response.errorCode != 0) {
+          alert("Error from U2F, code: " + response.errorCode);
+          return;
+        }
+        var mfaForm = document.getElementById('mfaForm');
+        var mfaResponse = document.getElementById('mfaSubmission');
+        mfaResponse.value = JSON.stringify(response);
+        mfaForm.submit();
+      });
+    }
+</script>
+<form action="<?= htmlentities($this->data['formTarget']); ?>" method="POST" id="mfaForm">
   
     <?php foreach ($this->data['formData'] as $name => $value): ?>
         <input type="hidden"
                name="<?= htmlentities($name); ?>"
                value="<?= htmlentities($value); ?>" />
     <?php endforeach; ?>
     
-    <p>Please insert your security key and press its button.</p>
+    <p id="mfaInstructions">Please insert your security key and press its button.</p>
     <p>
+        <input type="checkbox" name="rememberMe" id="rememberMe" value="true" checked="checked"/>
+        <label for="rememberMe">Remember this computer for 30 days</label>
+        <br />
         <input type="hidden" id="mfaSubmission" name="mfaSubmission" />
-        <button type="submit" name="submitMfa" id="submitMfa"
-                style="padding: 4px 8px;">Submit</button>
+        <input type="hidden" id="submitMfa" name="submitMfa" />
+
     </p>
+
+    <?php
+    if (count($this->data['mfaOptions']) > 1) {
+        ?>
+        <p>
+            Don't have your security key handy? You may also use:
+        <ul>
+            <?php
+            foreach ($this->data['mfaOptions'] as $mfaOpt) {
+                if ($mfaOpt['type'] != 'u2f'){
```

Comment:
If you wanted, you could change...
`($mfaOpt['type'] != 'u2f')`
... to... 
`($mfaOpt['id'] !== $this->data['mfaOption']['id'])`
... so that when we allow multiple U2F options, it will still show the other U2F options.

Reply:
true, I'll let the real implementation be done in the material theme, this was just for dev/test